webapp/components/features/FollowList/FollowList.spec.js

```diff
@@ -0,0 +1,150 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followingCount)
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.following.length)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['following']])
+        })
+      })
+
+      describe('of `followedBy`', () => {
+        it('uses the `followedBy` data on :user', () => {
+          const wrapper = Wrapper({ type: 'followedBy' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followedByCount)
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.followedBy.length)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['followedBy']])
+        })
+      })
+    })
+
+    describe('given no type', () => {
+      it('defaults type to `following`', () => {
+        const wrapper = Wrapper()
+        expect(wrapper.text()).toContain('profile.network.following')
+      })
+    })
+
+    describe('given a user', () => {
+      describe('without connections', () => {
+        it('displays the followingNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-31T11:56:26Z" title="Tuesday, March 31st 2020, 1:56:26 pm +02:00">Mar 31, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,133 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a user', () => {
+      describe('without connections', () => {
+        it('displays the followingNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser })
+          expect(wrapper.find('.nobody-message').text()).toBe(
+            `${noConnectionsUser.name} ${wrapper.vm.$t(`profile.network.followingNobody`)}`,
+          )
+        })
+
+        it('displays the followedByNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser, type: 'followedBy' })
+          expect(wrapper.find('.nobody-message').text()).toBe(
+            `${noConnectionsUser.name} ${wrapper.vm.$t(`profile.network.followedByNobody`)}`,
+          )
+        })
+      })
+
+      describe('with up to 7 loaded connections', () => {
+        let followingWrapper
+        let followedByWrapper
+        beforeAll(() => {
+          followingWrapper = Wrapper()
+          followedByWrapper = Wrapper({ type: 'followedBy' })
+        })
+
+        it(`renders the connections`, () => {
+          expect(followedByWrapper.findAll('.user-teaser').length).toEqual(user.followedBy.length)
+          expect(followingWrapper.findAll('.user-teaser').length).toEqual(user.following.length)
+        })
+
+        it(`has a button to load all remaining connections`, async () => {
+          followingWrapper.find('.base-button').trigger('click')
+          followedByWrapper.find('.base-button').trigger('click')
+          expect(followingWrapper.emitted('fetchAllConnections')).toBeTruthy()
+          expect(followedByWrapper.emitted('fetchAllConnections')).toBeTruthy()
```


```suggestion
          expect(followedByWrapper.emitted('fetchAllConnections')).toEqual(['followedBy'])
```
I think it would be something like this, no? I try to avoid `toBeTruthy` for more specific tests, wherever possible.
Also, we try to avoid multiple `expect`s in the same it block. Could you please make this two `it` blocks?

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-01T06:56:44Z" title="Wednesday, April 1st 2020, 8:56:44 am +02:00">Apr 1, 2020</time>_

---


Then let's test the type explicitly: daa8816